### PR TITLE
Localization in Polish, Danish, French, and German

### PR DESCRIPTION
**Pull request by LettyYeti**
With improvements to Italian and Finnish also.

### CODE REVIEW

File: src/asmcnc/apps/maintenance_app/widget_maintenance_z_lubrication_reminder.py [hunk 1]
```diff
@@ -145,6 +145,8 @@ def update_strings(self):
         self.time_since_srew_lubricated_label.text = self.l.get_bold("TIME SINCE LEAD SCREW LUBRICATED")
         self.hours_label = self.l.get_str("hours")
 
+        self.update_label_font_size(self.time_since_srew_lubricated_label)
```
Comment (LettyYeti) [inline src/asmcnc/apps/maintenance_app/widget_maintenance_z_lubrication_reminder.py:148]: just noticed the typo on this (srew). definitely my fault. if you get the chance to fix it (and it's related references) then please do, but it's not mission critical.

File: src/asmcnc/apps/start_up_sequence/data_consent_app/screens/wifi_and_data_consent_3.py [hunk 2]
```diff
@@ -218,7 +218,11 @@ def __init__(self, **kwargs):
 		self.update_strings()
 		self.set_checkbox_default()
 
-		self.scroll_privacy_notice.privacy_notice.source = self.privacy_notice_path + self.l.lang + '.rst'
+		# Can't seem to use non english letters for file source so filename is different
+		if self.l.lang == 'Français (FR)':
+			self.scroll_privacy_notice.privacy_notice.source = self.privacy_notice_path + 'Francais (FR).rst'
+		else:
+			self.scroll_privacy_notice.privacy_notice.source = self.privacy_notice_path + self.l.lang + '.rst'
```
Comment (LettyYeti) [inline src/asmcnc/apps/start_up_sequence/data_consent_app/screens/wifi_and_data_consent_3.py:226]: just realised that this is a duplicate of what's in update_strings() function anyway, so can likely be deleted. again, my fault originally

File: src/asmcnc/apps/start_up_sequence/warranty_app/screens/screen_warranty_registration_3.py [hunk 3]
```diff
@@ -301,6 +300,8 @@ def update_strings(self):
 		self.error_message_bottom.text = self.l.get_str("Stuck on this screen? Contact us at https://www.yetitool.com/support")
 		self.next_button.text = self.l.get_str("Next") + "..."
 
+		self.update_font_size(self.error_message_bottom)
+
 	def update_font_size(self, value):
-		if len(value.text) < 85: value.font_size = self.default_font_size
-		else: value.font_size = '18sp'
+		if len(value.text) < 100: value.font_size = self.default_font_size
+		else: value.font_size = '16sp'
```
Comment (LettyYeti) [inline src/asmcnc/apps/start_up_sequence/warranty_app/screens/screen_warranty_registration_3.py:307]: is this still ok for any phrases that are between 85 and 100 chars?

File: src/asmcnc/apps/start_up_sequence/warranty_app/screens/screen_warranty_registration_3.py [hunk 4]
```diff
@@ -177,7 +177,7 @@
 						size_hint: (None,None)
 						width: dp(291)
 						height: dp(79)
-						on_press: root.next_screen()
+						on_press: root.next_screen(False)
```
Comment (LettyYeti) [inline src/asmcnc/apps/start_up_sequence/warranty_app/screens/screen_warranty_registration_3.py:180]: what difference did this make?

File: src/asmcnc/apps/wifi_app/wifi_documentation/Francais (FR).rst [hunk 5]
```diff
@@ -1,74 +1,74 @@
...
+Connexion via un point d'accès mobile
+-------------------------------------
+
+Nous avons noté des problèmes de connexions sur les appareils Apple. Nous vous conseillons de vous connecter sur Android lorsque cela est possible.
+
+
+Appareils Apple
+~~~~~~~~~~~~~~~
+
+Pour vous connecter à un appareil Apple, essayez les étapes suivantes :
```
Comment (LettyYeti) [inline src/asmcnc/apps/wifi_app/wifi_documentation/Francais (FR).rst:60]: think there's an extra space here

File: src/asmcnc/apps/start_up_sequence/data_consent_app/screens/wifi_and_data_consent_3.py (hunk 2 above)
Comment (DennisKiselev) [reply src/asmcnc/apps/start_up_sequence/data_consent_app/screens/wifi_and_data_consent_3.py:226]: Not sure why I duplicated it, must've not noticed the update_strings function being called initially, will delete

File: src/asmcnc/apps/start_up_sequence/warranty_app/screens/screen_warranty_registration_3.py (hunk 4 above)
Comment (DennisKiselev) [reply src/asmcnc/apps/start_up_sequence/warranty_app/screens/screen_warranty_registration_3.py:180]: This screen has an extra argument in the next_screen function which allows it to distinguish between whether it is being called by pressing the button or by the auto scheduled one which is constantly being called, and is used to stop error messages being shown if the user did not manually attempt to proceed, however the argument was being defaulted to true which meant that error messages weren't being displayed

File: src/asmcnc/apps/wifi_app/wifi_documentation/Francais (FR).rst (hunk 5 above)
Comment (DennisKiselev) [reply src/asmcnc/apps/wifi_app/wifi_documentation/Francais (FR).rst:60]: Not sure what you mean here, could you clarify?

File: src/asmcnc/core_UI/sequence_alarm/alarm_manager.py [hunk 6]
```diff
@@ -21,7 +21,7 @@
 	"ALARM:4" : "Probe fail. Probe was not in the expected state before starting probe cycle.",
 	"ALARM:5" : "Probe fail. Tool did not contact the probe within the search distance.",
 	"ALARM:6" : "Homing fail: SmartBench was reset during active homing cycle.",
-	"ALARM:7" : "Homing fail: the interrupt bar was triggered during the homing cycle.",
+	"ALARM:7" : "Homing fail: the stop bar was triggered during the homing cycle.",
```
[the remark this answers is not in the document]
Comment (DennisKiselev) [reply src/asmcnc/core_UI/sequence_alarm/alarm_manager.py:24]: Will also update error:7 on screen_check_job which also says stop bar then

File: src/asmcnc/apps/wifi_app/wifi_documentation/Francais (FR).rst (hunk 5 above)
Comment (LettyYeti) [reply src/asmcnc/apps/wifi_app/wifi_documentation/Francais (FR).rst:60]: between suivantes and the ":", seems there's an extra space

File: src/asmcnc/apps/systemTools_app/screens/screen_support_menu.py [hunk 7]
```diff
@@ -148,8 +148,4 @@ def update_font_size(self, value):
         elif len(value.text) > 15: 
             value.font_size = self.default_font_size - 2
         if len(value.text) > 19: 
-            value.font_size = self.default_font_size - 4
-        if len(value.text) > 22: 
-            value.font_size = self.default_font_size - 5
-        if len(value.text) > 25: 
-            value.font_size = self.default_font_size - 6
```
Comment (LettyYeti) [inline src/asmcnc/apps/systemTools_app/screens/screen_support_menu.py:155]: were these just unnecessary?

Comment (DennisKiselev) [reply src/asmcnc/apps/systemTools_app/screens/screen_support_menu.py:155]: Just had a look and it all looks fine, I'm guessing they were just making the text too small to be easily legible

**Pull request merged**